### ISSUE TITLE
ResourceExhaustedError  -- Allocator (GPU_0_bfc) ran out of memory trying to allocate 37.67MiB

### ISSUE THREAD

**Issue by GPU-Shuttle** [DeepLabCut/DeepLabCut#1144]
As per [instructions](https://github.com/DeepLabCut/DeepLabCut) on how to use Effnets, I installed v2.1.10 and passed efficientnet-b6  when creating the trainingset: "deeplabcut.create_training_dataset(path_config, net_type='efficientnet-b6')". all good up to this stage, however, DLC returns an error at the training step: "deeplabcut.train_network(path_config)", which is copied below.
Any tips would be appreciated.

```python
-------------------------------------

W tensorflow/core/common_runtime/bfc_allocator.cc:267] Allocator (GPU_0_bfc) ran out of memory trying to allocate 37.67MiB.  Current allocation summary follows.

2021-03-15 11:06:07.202518: W tensorflow/core/framework/op_kernel.cc:1401] OP_REQUIRES failed at transpose_op.cc:199 : Resource exhausted: OOM when allocating tensor with shape[1,2064,52,92] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
---------------------------------------------------------------------------
ResourceExhaustedError                    Traceback (most recent call last)
~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _do_call(self, fn, *args)
   1333     try:
-> 1334       return fn(*args)
   1335     except errors.OpError as e:

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _run_fn(feed_dict, fetch_list, target_list, options, run_metadata)
   1318       return self._call_tf_sessionrun(
-> 1319           options, feed_dict, fetch_list, target_list, run_metadata)
   1320

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _call_tf_sessionrun(self, options, feed_dict, fetch_list, target_list, run_metadata)
   1406         self._session, options, feed_dict, fetch_list, target_list,
-> 1407         run_metadata)
   1408

ResourceExhaustedError: OOM when allocating tensor with shape[1,2064,52,92] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
         [[{{node efficientnet/efficientnet-b6/model/blocks_41/depthwise_conv2d/depthwise-0-TransposeNHWCToNCHW-LayoutOptimizer}}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

         [[{{node absolute_difference/weighted_loss/assert_broadcastable/is_valid_shape/has_valid_nonscalar_shape/has_invalid_dims/concat}}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.


During handling of the above exception, another exception occurred:

ResourceExhaustedError                    Traceback (most recent call last)
<ipython-input-4-18dc6009ffca> in <module>
----> 1 deeplabcut.train_network(path_config)

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/training.py in train_network(config, shuffle, trainingsetindex, max_snapshots_to_keep, displayiters, saveiters, maxiters, allow_growth, gputouse, autotune, keepdeconvweights, modelprefix)
    191
    192     except BaseException as e:
--> 193         raise e
    194     finally:
    195         os.chdir(str(start_path))

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/training.py in train_network(config, shuffle, trainingsetindex, max_snapshots_to_keep, displayiters, saveiters, maxiters, allow_growth, gputouse, autotune, keepdeconvweights, modelprefix)
    187                 max_to_keep=max_snapshots_to_keep,
    188                 keepdeconvweights=keepdeconvweights,
--> 189                 allow_growth=allow_growth,
    190             )  # pass on path and file name for pose_cfg.yaml!
    191

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/train.py in train(config_yaml, displayiters, saveiters, maxiters, max_to_keep, keepdeconvweights, allow_growth)
    276         [_, loss_val, summary] = sess.run(
    277             [train_op, total_loss, merged_summaries],
--> 278             feed_dict=dict,
    279         )
    280         cum_loss += loss_val

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in run(self, fetches, feed_dict, options, run_metadata)
    927     try:
    928       result = self._run(None, fetches, feed_dict, options_ptr,
--> 929                          run_metadata_ptr)
    930       if run_metadata:
    931         proto_data = tf_session.TF_GetBuffer(run_metadata_ptr)

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _run(self, handle, fetches, feed_dict, options, run_metadata)
   1150     if final_fetches or final_targets or (handle and feed_dict_tensor):
   1151       results = self._do_run(handle, final_targets, final_fetches,
-> 1152                              feed_dict_tensor, options, run_metadata)
   1153     else:
   1154       results = []

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _do_run(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)
   1326     if handle is None:
   1327       return self._do_call(_run_fn, feeds, fetches, targets, options,
-> 1328                            run_metadata)
   1329     else:
   1330       return self._do_call(_prun_fn, handle, feeds, fetches)

~/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/tensorflow/python/client/session.py in _do_call(self, fn, *args)
   1346           pass
   1347       message = error_interpolation.interpolate(message, self._graph)
-> 1348       raise type(e)(node_def, op, message)
   1349
   1350   def _extend_graph(self):

ResourceExhaustedError: OOM when allocating tensor with shape[1,2064,52,92] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
         [[{{node efficientnet/efficientnet-b6/model/blocks_41/depthwise_conv2d/depthwise-0-TransposeNHWCToNCHW-LayoutOptimizer}}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

         [[node absolute_difference/weighted_loss/assert_broadcastable/is_valid_shape/has_valid_nonscalar_shape/has_invalid_dims/concat (defined at /home/user/anaconda3/envs/DLC-GPU/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/nnet/losses.py:45) ]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
```

**Comment by jeylau** [comment]
Duplicate of #613 #133.
Your GPU is likely too small: you could try to reduce 'global_scale' in the pose_cfg.yaml under dlc_models/././train, or downsample your videos first using `deeplabcut.DownSampleVideo`

**Comment by GPU-Shuttle** [comment]
thanks, will try these.
Although, FYI, I have previously trained a Resnet152 using the same videos successfully on the same system (gpu: TITAN V, 12GB)

**Comment by MMathisLab** [comment]
`Allocator (GPU_0_bfc) ran out of memory trying to allocate 37.67MiB` ; <<< the batch size is perhaps too high; if you adjust it down, please also be mindful of the learning rates... 

Also, please in the future use the template, as it really helps use parse these issues efficiently.

**Comment by backyardbiomech** [comment]
I ran into the same problem when I tried a quick switch to efficient nets in one my existing projects. My guess is that the pretrained model is larger than for resnet50 (which I was using), which leaves less room in the GPU memory for image loading, so I had to lower my batch size (or cropping/downsampling should also work). I didn't spend to much time on it because I was trying out of curiosity.
